mcpscanner/core/static_analysis/dataflow/treesitter_classic.py

```
@@ -0,0 +1,538 @@
+        reaching = [d for d in fact.defs if d.var == name]
+        if any(d.is_parameter for d in reaching):
+            return True
+    return False
+
+
+def _normalize_expr(node: Node, source_bytes: bytes) -> str:
+    text = _node_text(node, source_bytes).strip()
+    return text[:200] if text else ""
+
+
+def _expr_uses_vars(expr_str: str, vars: set[str]) -> bool:
```

Ruff (0.16.3)

[error] Function argument `vars` is shadowing a Python builtin

(A002)